src/s3path.jl

```diff
@@ -174,7 +174,12 @@ function Base.isdir(fp::S3Path)
     end
 
     objects = s3_list_objects(fp.config, fp.bucket, key; max_items=1)
-    return !isempty(objects)
+
+    # `objects` is a `Channel`, so we call iterate to see if there are any objects that
```

Comment:
Would `iterate` not be checking the first item and returning a tuple of itself and the init state or `nothing`?

> https://docs.julialang.org/en/v1/manual/interfaces/

Reply:
That's correct. Since this function just wants to know if there are any items that match we're just checking that it doesn't return `nothing`.